osquery/filesystem/filesystem.cpp

```diff
@@ -131,15 +143,15 @@ Status readFile(const fs::path& path,
     try {
       return Status(0, fs::canonical(path, ec).string());
     } catch (const boost::filesystem::filesystem_error& err) {
-      return Status(1, err.what());
+      return Status::failure(err.what());
     }
   }
 
   PlatformTime times;
   handle.fd->getFileTimes(times);
 
   off_t total_bytes = 0;
-  if (file_size == 0 || block_size > 0) {
+  if (handle.blocking_io) {
     // Reset block size to a sane minimum.
     block_size = (block_size < 4096) ? 4096 : block_size;
```

Comment:
I suspect any block size less than 1MB will be inefficient from an IO perspective for reading files. This is from experiments I ran back in 2006. This link has some data that may explain when 4KB block sizes are preferred over 1MB block sizes and cases where 1MB is preferred: https://medium.com/@duhroach/the-impact-of-blocksize-on-persistent-disk-performance-7e50a85b2647

Reply:
I think it highly depends on the files, the disk, the filesystem etc.
This is just a sane minimum, one is free to use a higher value when using the readFile API with files that are large enough where this makes sense.
Although I could get behind maybe changing it here: https://github.com/osquery/osquery/blob/a9770451c527eba4cf321fe23c5306beff6ac20b/osquery/filesystem/filesystem.cpp#L189.

That been said, I feel this is a different matter that needs its own issue and PR :P.